lib/geometry/polygon.nit

```diff
@@ -0,0 +1,428 @@
+# This file is part of NIT (http://www.nitlanguage.org).
+#
+# Copyright 2014 Romain Chanoir <romain.chanoir@viacesi.fr>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# Convex Polygons manipulations
+module polygon
+
+import points_and_lines
+
+
+# Convex Polygon class
+class ConvexPolygon
+
+	# Vertices of this polygon
+	var points = new Array[Point[Float]]
+
+	# Init this polygon with a list of at least 3 vertices
```

Comment:
I feel that the comment is not clear enough, maybe add a REQUIRE instead of the 3 vertices part